### PR TITLE
[SPARK-35700][SQL] Read char/varchar orc table with created and written by external systems

### PR DESCRIPTION
**Pull request by yaooqinn**
### What changes were proposed in this pull request?

The char/varchar type should be mapped to orc's string type too, see https://orc.apache.org/docs/types.html

### Why are the changes needed?

fix a regression

### Does this PR introduce _any_ user-facing change?

no

### How was this patch tested?

new tests

### CODE REVIEW

**Review by dongjoon-hyun — COMMENTED** [review]
The newly add UT seems to fail. Could you check it, @yaooqinn ?

**Comment by dongjoon-hyun** [comment]
Thank you for updating, @yaooqinn . Can we remove `[WIP]` from the title?

File: sql/hive/src/test/scala/org/apache/spark/sql/HiveCharVarcharTestSuite.scala [hunk 1]
```diff
@@ -17,7 +17,9 @@
 
 package org.apache.spark.sql
 
+import org.apache.spark.sql.catalyst.catalog.ExternalCatalogWithListener
```
Comment (dongjoon-hyun) [inline sql/hive/src/test/scala/org/apache/spark/sql/HiveCharVarcharTestSuite.scala:20]:
```suggestion
```

File: sql/hive/src/test/scala/org/apache/spark/sql/HiveCharVarcharTestSuite.scala [hunk 2]
```diff
@@ -17,7 +17,9 @@
...
 import org.apache.spark.sql.execution.command.CharVarcharDDLTestBase
+import org.apache.spark.sql.hive.HiveExternalCatalog
```
Comment (dongjoon-hyun) [inline sql/hive/src/test/scala/org/apache/spark/sql/HiveCharVarcharTestSuite.scala:22]:
```suggestion
```

File: sql/hive/src/test/scala/org/apache/spark/sql/HiveCharVarcharTestSuite.scala [hunk 3]
```diff
@@ -50,6 +52,20 @@ class HiveCharVarcharTestSuite extends CharVarcharTestSuite with TestHiveSinglet
       assert(rest.contains("CHAR(5)"))
     }
   }
+
+  test("SPARK-35700: Read char/varchar orc table with created and written by external systems") {
+    val hiveClient = spark.sharedState
+      .externalCatalog.asInstanceOf[ExternalCatalogWithListener]
+      .unwrapped
+      .asInstanceOf[HiveExternalCatalog].client
```
Comment (dongjoon-hyun) [inline sql/hive/src/test/scala/org/apache/spark/sql/HiveCharVarcharTestSuite.scala:60]:
```suggestion
```

**Review by dongjoon-hyun — COMMENTED** [review]
`hiveClient` is already provided `TestHiveSingleton`. Let's remove it.

**Review by dongjoon-hyun — APPROVED** [review]
+1, LGTM.

Actually, this is Hive type.
- https://github.com/apache/hive/blob/master/storage-api/src/java/org/apache/hadoop/hive/ql/io/sarg/PredicateLeaf.java#L52

**Review by HyukjinKwon — COMMENTED** [review]
lgtm2

**Pull request closed without merging**